test/podFailover/controller/cmd/main.go

```diff
@@ -204,6 +211,22 @@ func deletePod(ctx context.Context, podName string, namespace string, clientset
 	return waitForPodToBeDeleted(ctx, podName, namespace, clientset)
 }
 
+func waitForPodCreated(ctx context.Context, clientset *kubernetes.Clientset, pod v1.Pod, namespace string) error {
+	backoff := wait.Backoff{Duration: 1 * time.Second, Factor: 2.0, Steps: 5}
+	return retry.OnError(
+		backoff,
+		func(err error) bool {
+			return strings.Contains(err.Error(), "not found")
```

Comment:
This should be `return errors.IsNotFound(err)`. See [errors.IsNotFound](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound).